src/propagators/zipkin_b3_text_map_codec.js

```diff
@@ -0,0 +1,176 @@
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+import * as constants from '../constants.js';
+import Metrics from '../metrics/metrics.js';
+import NoopMetricFactory from '../metrics/noop/metric_factory';
+import SpanContext from '../span_context.js';
+import Utils from '../util.js';
+import { parseCommaSeparatedBaggage } from '../propagators/baggage';
+
+const ZIPKIN_PARENTSPAN_HEADER = 'x-b3-parentspanid';
+const ZIPKIN_SPAN_HEADER = 'x-b3-spanid';
+const ZIPKIN_TRACE_HEADER = 'x-b3-traceid';
+const ZIPKIN_SAMPLED_HEADER = 'x-b3-sampled';
+const ZIPKIN_FLAGS_HEADER = 'x-b3-flags';
+
+export default class ZipkinB3TextMapCodec {
+    _urlEncoding: boolean;
+    _baggagePrefix: string;
+    _metrics: any;
+
+    constructor(options: any = {}) {
+        this._urlEncoding = !!options.urlEncoding;
+        this._baggagePrefix = options.baggagePrefix || constants.TRACER_BAGGAGE_HEADER_PREFIX;
+        this._baggagePrefix = this._baggagePrefix.toLowerCase();
+        this._metrics = options.metrics || new Metrics(new NoopMetricFactory());
+    }
+
+    _encodeValue(value: string): string {
+        if (this._urlEncoding) {
+            return encodeURIComponent(value);
+        }
+
+        return value;
+    }
+
+    _decodeValue(value: string): string {
+        // only use url-decoding if there are meta-characters '%'
+        if (this._urlEncoding && value.indexOf('%') > -1) {
+            return this._decodeURIValue(value);
+        }
+
+        return value;
+    }
+
+    _isValidZipkinId(value: string): boolean {
+        // Validates a zipkin trace/spanID by attempting to parse it as a
+        // string of hex digits. This "validation" is not entirely rigorous,
+        // but equivalent to what is performed in the TextMapCodec.
+        //
+        // Note: due to the way parseInt works, this does not guarantee that
+        // the string is composed *entirely* of hex digits.
+        //
+        // > If parseInt encounters a character that is not a numeral in the
+        // > specified radix, it ignores it and all succeeding characters and
+        // > returns the integer value parsed up to that point.
+        //
+        // Note: The Number type in JS cannot represent the full range of 64bit
+        // unsigned ints, so using parseInt() on strings representing 64bit hex
+        // numbers only returns an approximation of the actual value.
+        // Fortunately, we do not depend on the returned value, we are only
+        // using it to validate that the string is a valid hex number (which is
+        // faster than doing it manually).  We cannot use
+        // Int64(numberValue).toBuffer() because it throws exceptions on bad
+        // strings.
+        if (!value) {
+            return true
```

Comment:
return false?

Reply:
I return true here since an empty string is technically a valid value to find in a zipkin header. (`ParentSpanId` can be absent [for instance](https://github.com/openzipkin/b3-propagation#parentspanid)).  I'm only using the return value of this method in the `extract`method, to determine if any invalid header values were found.